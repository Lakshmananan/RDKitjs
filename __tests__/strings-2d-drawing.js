'use strict';

var RDKit = require('..');

beforeEach(() => RDKit.load());

test('smilewrite', function() {
  var smi = 'CCCCCOC(CO)';
  var mol = RDKit.Molecule.smilesToMol(smi);
  expect(mol.molToSmiles()).toBe('CCCCCOCCO');
  mol.delete();
});

test('sdwriteConfs', function() {
  var smi = 'CCCCCOC(CO)';
  var mol = RDKit.Molecule.smilesToMol(smi);
  mol.addHs();
  mol.EmbedMultipleConfsarg(3, 1000, 2015);
  expect(mol.sdwriteConfs()).toMatchSnapshot();
  mol.delete();
});

test('2D generation', function() {
  var smi = 'CCCCCOC(CO)';
  var mol = RDKit.Molecule.smilesToMol(smi);
  mol.compute2DCoords();
  expect(mol.sdwriteConfs()).toMatchSnapshot();
  mol.delete();
});

test('2D Drawing', function() {
  var smi = 'CCCCCOC(CO)';
  var mol = RDKit.Molecule.smilesToMol(smi);
  expect(mol.Drawing2D()).toMatchSnapshot();
  mol.delete();
});
