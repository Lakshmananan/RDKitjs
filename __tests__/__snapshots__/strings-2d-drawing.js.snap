// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`2D Drawing 1`] = `
"<?xml version='1.0' encoding='iso-8859-1'?>
<svg:svg version='1.1' baseProfile='full'
              xmlns:svg='http://www.w3.org/2000/svg'
                      xmlns:rdkit='http://www.rdkit.org/xml'
                      xmlns:xlink='http://www.w3.org/1999/xlink'
                  xml:space='preserve'
width='300px' height='300px' >
<svg:rect style='opacity:1.0;fill:#FFFFFF;stroke:none' width='300' height='300' x='0' y='0'> </svg:rect>
<svg:path d='M 13.6364,159.057 45.0092,140.943' style='fill:none;fill-rule:evenodd;stroke:#000000;stroke-width:2px;stroke-linecap:butt;stroke-linejoin:miter;stroke-opacity:1' />
<svg:path d='M 45.0092,140.943 76.3821,159.057' style='fill:none;fill-rule:evenodd;stroke:#000000;stroke-width:2px;stroke-linecap:butt;stroke-linejoin:miter;stroke-opacity:1' />
<svg:path d='M 76.3821,159.057 107.755,140.943' style='fill:none;fill-rule:evenodd;stroke:#000000;stroke-width:2px;stroke-linecap:butt;stroke-linejoin:miter;stroke-opacity:1' />
<svg:path d='M 107.755,140.943 139.128,159.057' style='fill:none;fill-rule:evenodd;stroke:#000000;stroke-width:2px;stroke-linecap:butt;stroke-linejoin:miter;stroke-opacity:1' />
<svg:path d='M 139.128,159.057 151.995,151.628' style='fill:none;fill-rule:evenodd;stroke:#000000;stroke-width:2px;stroke-linecap:butt;stroke-linejoin:miter;stroke-opacity:1' />
<svg:path d='M 151.995,151.628 164.862,144.199' style='fill:none;fill-rule:evenodd;stroke:#FF0000;stroke-width:2px;stroke-linecap:butt;stroke-linejoin:miter;stroke-opacity:1' />
<svg:path d='M 176.14,144.199 189.007,151.628' style='fill:none;fill-rule:evenodd;stroke:#FF0000;stroke-width:2px;stroke-linecap:butt;stroke-linejoin:miter;stroke-opacity:1' />
<svg:path d='M 189.007,151.628 201.873,159.057' style='fill:none;fill-rule:evenodd;stroke:#000000;stroke-width:2px;stroke-linecap:butt;stroke-linejoin:miter;stroke-opacity:1' />
<svg:path d='M 201.873,159.057 233.246,140.943' style='fill:none;fill-rule:evenodd;stroke:#000000;stroke-width:2px;stroke-linecap:butt;stroke-linejoin:miter;stroke-opacity:1' />
<svg:path d='M 233.246,140.943 243.704,146.981' style='fill:none;fill-rule:evenodd;stroke:#000000;stroke-width:2px;stroke-linecap:butt;stroke-linejoin:miter;stroke-opacity:1' />
<svg:path d='M 243.704,146.981 254.162,153.019' style='fill:none;fill-rule:evenodd;stroke:#FF0000;stroke-width:2px;stroke-linecap:butt;stroke-linejoin:miter;stroke-opacity:1' />
<svg:text x='164.862' y='146.981' style='font-size:12px;font-style:normal;font-weight:normal;fill-opacity:1;stroke:none;font-family:sans-serif;text-anchor:start;fill:#FF0000' ><svg:tspan>O</svg:tspan></svg:text>
<svg:text x='253.747' y='165.094' style='font-size:12px;font-style:normal;font-weight:normal;fill-opacity:1;stroke:none;font-family:sans-serif;text-anchor:start;fill:#FF0000' ><svg:tspan>OH</svg:tspan></svg:text>
</svg:svg>
"
`;

exports[`2D generation 1`] = `
"
     RDKit          2D

  9  8  0  0  0  0  0  0  0  0999 V2000
    0.0000    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.2990    0.7500    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    2.5981   -0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    3.8971    0.7500    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    5.1962   -0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    6.4952    0.7500    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0
    7.7942   -0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    9.0933    0.7500    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   10.3923   -0.0000    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  1  0
  2  3  1  0
  3  4  1  0
  4  5  1  0
  5  6  1  0
  6  7  1  0
  7  8  1  0
  8  9  1  0
M  END
$$$$
"
`;

exports[`sdwriteConfs 1`] = `
"
     RDKit          3D

 25 24  0  0  0  0  0  0  0  0999 V2000
    3.9840    0.1254   -0.2476 C   0  0  0  0  0  0  0  0  0  0  0  0
    2.4817   -0.0129   -0.4820 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.8254    0.1570    0.8479 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.3452    0.0574    0.8647 C   0  0  0  0  0  0  0  0  0  0  0  0
   -0.2735   -0.7012   -0.2616 C   0  0  0  0  0  0  0  0  0  0  0  0
   -1.3948    0.0027   -0.7319 O   0  0  0  0  0  0  0  0  0  0  0  0
   -2.3754   -0.0696    0.2499 C   0  0  0  0  0  0  0  0  0  0  0  0
   -3.7151    0.4002   -0.2556 C   0  0  0  0  0  0  0  0  0  0  0  0
   -4.6855   -0.2127    0.5440 O   0  0  0  0  0  0  0  0  0  0  0  0
    4.3140   -0.7630    0.3470 H   0  0  0  0  0  0  0  0  0  0  0  0
    4.5389    0.2440   -1.1883 H   0  0  0  0  0  0  0  0  0  0  0  0
    4.0917    0.9917    0.4463 H   0  0  0  0  0  0  0  0  0  0  0  0
    2.3664   -1.0336   -0.9132 H   0  0  0  0  0  0  0  0  0  0  0  0
    2.2266    0.7769   -1.2202 H   0  0  0  0  0  0  0  0  0  0  0  0
    2.1107    1.1721    1.2215 H   0  0  0  0  0  0  0  0  0  0  0  0
    2.3086   -0.5571    1.5514 H   0  0  0  0  0  0  0  0  0  0  0  0
   -0.1332    1.0649    0.9630 H   0  0  0  0  0  0  0  0  0  0  0  0
    0.0585   -0.4749    1.8083 H   0  0  0  0  0  0  0  0  0  0  0  0
   -0.6698   -1.6968    0.0629 H   0  0  0  0  0  0  0  0  0  0  0  0
    0.3651   -0.8524   -1.1388 H   0  0  0  0  0  0  0  0  0  0  0  0
   -2.1166    0.4793    1.1669 H   0  0  0  0  0  0  0  0  0  0  0  0
   -2.4950   -1.1513    0.5010 H   0  0  0  0  0  0  0  0  0  0  0  0
   -3.8321    1.5065   -0.1774 H   0  0  0  0  0  0  0  0  0  0  0  0
   -3.8780    0.1498   -1.3316 H   0  0  0  0  0  0  0  0  0  0  0  0
   -5.4477    0.3974    0.6318 H   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  1  0
  2  3  1  0
  3  4  1  0
  4  5  1  0
  5  6  1  0
  6  7  1  0
  7  8  1  0
  8  9  1  0
  1 10  1  0
  1 11  1  0
  1 12  1  0
  2 13  1  0
  2 14  1  0
  3 15  1  0
  3 16  1  0
  4 17  1  0
  4 18  1  0
  5 19  1  0
  5 20  1  0
  7 21  1  0
  7 22  1  0
  8 23  1  0
  8 24  1  0
  9 25  1  0
M  END
$$$$

     RDKit          3D

 25 24  0  0  0  0  0  0  0  0999 V2000
   -3.9848    0.3867   -0.4439 C   0  0  0  0  0  0  0  0  0  0  0  0
   -2.5115    0.7744   -0.3701 C   0  0  0  0  0  0  0  0  0  0  0  0
   -1.8420   -0.2667    0.4844 C   0  0  0  0  0  0  0  0  0  0  0  0
   -0.7777   -0.9402   -0.3676 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.5651   -0.6601    0.2538 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.4788   -0.1226   -0.6508 O   0  0  0  0  0  0  0  0  0  0  0  0
    2.7468   -0.4783   -0.2026 C   0  0  0  0  0  0  0  0  0  0  0  0
    3.4842    0.7826    0.2082 C   0  0  0  0  0  0  0  0  0  0  0  0
    4.8421    0.5349   -0.0503 O   0  0  0  0  0  0  0  0  0  0  0  0
   -3.9794   -0.7138   -0.6485 H   0  0  0  0  0  0  0  0  0  0  0  0
   -4.4952    0.9656   -1.2331 H   0  0  0  0  0  0  0  0  0  0  0  0
   -4.4249    0.5255    0.5621 H   0  0  0  0  0  0  0  0  0  0  0  0
   -2.0972    0.7996   -1.4101 H   0  0  0  0  0  0  0  0  0  0  0  0
   -2.4076    1.8040    0.0178 H   0  0  0  0  0  0  0  0  0  0  0  0
   -1.3847    0.2562    1.3599 H   0  0  0  0  0  0  0  0  0  0  0  0
   -2.5462   -1.0109    0.9036 H   0  0  0  0  0  0  0  0  0  0  0  0
   -0.9971   -2.0369   -0.3822 H   0  0  0  0  0  0  0  0  0  0  0  0
   -0.8284   -0.6078   -1.4156 H   0  0  0  0  0  0  0  0  0  0  0  0
    0.4756    0.0879    1.0816 H   0  0  0  0  0  0  0  0  0  0  0  0
    0.9301   -1.6111    0.7109 H   0  0  0  0  0  0  0  0  0  0  0  0
    3.2646   -0.9441   -1.0724 H   0  0  0  0  0  0  0  0  0  0  0  0
    2.7159   -1.1803    0.6581 H   0  0  0  0  0  0  0  0  0  0  0  0
    3.1126    1.6451   -0.3759 H   0  0  0  0  0  0  0  0  0  0  0  0
    3.3004    0.9868    1.2741 H   0  0  0  0  0  0  0  0  0  0  0  0
    5.3607    1.0236    0.6249 H   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  1  0
  2  3  1  0
  3  4  1  0
  4  5  1  0
  5  6  1  0
  6  7  1  0
  7  8  1  0
  8  9  1  0
  1 10  1  0
  1 11  1  0
  1 12  1  0
  2 13  1  0
  2 14  1  0
  3 15  1  0
  3 16  1  0
  4 17  1  0
  4 18  1  0
  5 19  1  0
  5 20  1  0
  7 21  1  0
  7 22  1  0
  8 23  1  0
  8 24  1  0
  9 25  1  0
M  END
$$$$

     RDKit          3D

 25 24  0  0  0  0  0  0  0  0999 V2000
   -3.7097    0.3392   -0.4066 C   0  0  0  0  0  0  0  0  0  0  0  0
   -2.8809    0.4861    0.8390 C   0  0  0  0  0  0  0  0  0  0  0  0
   -1.6251   -0.3382    0.7994 C   0  0  0  0  0  0  0  0  0  0  0  0
   -0.5627    0.2268   -0.1067 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.4034   -0.9171   -0.3938 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.6352   -0.4745   -0.8353 O   0  0  0  0  0  0  0  0  0  0  0  0
    2.4370   -0.0264    0.2199 C   0  0  0  0  0  0  0  0  0  0  0  0
    3.9086   -0.1163   -0.1239 C   0  0  0  0  0  0  0  0  0  0  0  0
    4.6121    0.4214    0.9612 O   0  0  0  0  0  0  0  0  0  0  0  0
   -4.7582    0.6227   -0.1885 H   0  0  0  0  0  0  0  0  0  0  0  0
   -3.3440    0.9860   -1.2185 H   0  0  0  0  0  0  0  0  0  0  0  0
   -3.6474   -0.7086   -0.7461 H   0  0  0  0  0  0  0  0  0  0  0  0
   -2.7227    1.5434    1.0810 H   0  0  0  0  0  0  0  0  0  0  0  0
   -3.5028    0.0682    1.6716 H   0  0  0  0  0  0  0  0  0  0  0  0
   -1.2215   -0.4300    1.8193 H   0  0  0  0  0  0  0  0  0  0  0  0
   -1.9214   -1.3562    0.4416 H   0  0  0  0  0  0  0  0  0  0  0  0
   -0.9477    0.5656   -1.0784 H   0  0  0  0  0  0  0  0  0  0  0  0
    0.0114    1.0298    0.3870 H   0  0  0  0  0  0  0  0  0  0  0  0
   -0.0193   -1.5675   -1.1996 H   0  0  0  0  0  0  0  0  0  0  0  0
    0.4740   -1.5333    0.5365 H   0  0  0  0  0  0  0  0  0  0  0  0
    2.2174    1.0460    0.4175 H   0  0  0  0  0  0  0  0  0  0  0  0
    2.2851   -0.6215    1.1461 H   0  0  0  0  0  0  0  0  0  0  0  0
    4.1682    0.5141   -1.0047 H   0  0  0  0  0  0  0  0  0  0  0  0
    4.2679   -1.1450   -0.2547 H   0  0  0  0  0  0  0  0  0  0  0  0
    4.4432    1.3854    1.0113 H   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  1  0
  2  3  1  0
  3  4  1  0
  4  5  1  0
  5  6  1  0
  6  7  1  0
  7  8  1  0
  8  9  1  0
  1 10  1  0
  1 11  1  0
  1 12  1  0
  2 13  1  0
  2 14  1  0
  3 15  1  0
  3 16  1  0
  4 17  1  0
  4 18  1  0
  5 19  1  0
  5 20  1  0
  7 21  1  0
  7 22  1  0
  8 23  1  0
  8 24  1  0
  9 25  1  0
M  END
$$$$
"
`;
